spec/nandi/config_spec.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "nandi/config"
+
+RSpec.describe Nandi::Config do
+  subject(:config) { described_class.new }
+
+  before do
+    Nandi.instance_variable_set(:@config, nil) # Reset config
+    config.lockfile_directory = "db"
+  end
+
+  context "multi-database integration" do
+    before do
+      config.register_database(
+        :primary,
+        migration_directory: primary_migration_directory,
+        output_directory: primary_output_directory,
+      )
+      config.register_database(
+        :analytics,
+        migration_directory: analytics_migration_directory,
+        output_directory: analytics_output_directory,
+      )
+    end
```

Comment:
Do we want to test registering `default: true` and the logic associated with that? Happy to be done as a follow-up